src/main/scala/dsls/regex/RegularExpression.scala

```diff
@@ -11,6 +11,39 @@ package dsls.regex
 abstract class RegularExpression {
   /** returns true if the given string matches this regular expression */
   def matches(string: String) = RegexMatcher.matches(string, this)
+
+  /** returns a regular expression matching either one sub-expression or another */  
+  def ||(other: RegularExpression) = Union(this, other)
+
+  /** returns a regular expression matching one sub-expression followed by another */  
+  def ~(other: RegularExpression) = Concat(this, other)
+
+  /** returns a regular expression matching zero or more repetitions of the expression */  
+  def <*> = Star(this)
+
+  /** returns a regular expression matching one or more repetitions of the expression */  
+  def <+> = {
+  	val zeroOrMoreDigits = Star(this)
+  	Concat(this, zeroOrMoreDigits)
+  }
+
+  /** returns a regular expression matching n repetitions of the expression */  
+  def apply(n: Int): RegularExpression = {
+  	if(n==0) EPSILON
+  	else Concat(this, this{n-1})
+  }
+
+
+}
+
+object RegularExpression{
+	implicit def charToLiteral(c : Char) = Literal(c)
+	implicit def stringToRegex(s: String): RegularExpression = charListHelper(s.toList)
+
+	implicit def charListHelper(list: List[Char]): RegularExpression = list match{
+		case List() => EPSILON
+		case c::rest => Concat(c,rest)
```

Comment:
I also converted a string into a list of chars, but used folding to make the regex. I think some comments for `charListHelper` would clarify the algorithm